javascript/ql/src/semmle/javascript/StandardLibrary.qll

```diff
@@ -146,3 +146,37 @@ class StringReplaceCall extends DataFlow::MethodCallNode {
     )
   }
 }
+
+/**
+ * A call to `String.prototype.split`.
+ *
+ * We heuristically include any call to a method called `split`, provided it either
+ * has exactly one arguments, or local data flow suggests that the receiver may be a string.
+ */
+class StringSplitCall extends DataFlow::MethodCallNode {
+  StringSplitCall() {
+    this.getMethodName() = "split" and
+    (getNumArgument() = 1 or getReceiver().mayHaveStringValue(_))
+  }
+
+  /**
+   * Gets a string that determines where the string is split.
+   */
+  string getSplitAt() {
```

Comment:
I didn't want to block your PR on naming issues like this, but since we're at it:
```suggestion
  string getDelimiter() {
```

Reply:
```suggestion
  string getSeparator() {
```
(matches the MDN signature above)

Reply:
Either is fine with me